staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go

```diff
@@ -157,6 +158,8 @@ func NewDrainCmdOptions(f cmdutil.Factory, ioStreams genericiooptions.IOStreams)
 		},
 	}
 	o.drainer.OnPodDeletedOrEvicted = o.onPodDeletedOrEvicted
```

Comment:
I'd expect drain command not to use the old methods, but only the new hooks. Especially that if you have both `OnPodDeletedOrEvicted` and `OnPodDeletionOrEvictionFinished` defined, only the latter is being used, right?

Reply:
With your current implementation if I invoke `kubectl drain -v=2` you'll get pod printed more than necessary. I don't think that's necessary, I'd like to maintain the current behavior of the drain command, so you need to make sure that whatever changes under the covers the command invocation stays the same.

Reply:
checking verbosity level while creating `NewDrainCmdOptions` should fix this right? like this:

```
func NewDrainCmdOptions(f cmdutil.Factory, ioStreams genericiooptions.IOStreams) *DrainCmdOptions {
	o := &DrainCmdOptions{
		PrintFlags: genericclioptions.NewPrintFlags("drained").WithTypeSetter(scheme.Scheme),
		IOStreams:  ioStreams,
		drainer: &drain.Helper{
			GracePeriodSeconds: -1,
			Out:                ioStreams.Out,
			ErrOut:             ioStreams.ErrOut,
			ChunkSize:          cmdutil.DefaultChunkSize,
		},
	}
	if !klog.V(2).Enabled() {
		o.drainer.OnPodDeletedOrEvicted = o.onPodDeletedOrEvicted
	} else {
		o.drainer.OnPodDeletionOrEvictionFinished = o.onPodDeletionOrEvictionFinished
		o.drainer.OnPodDeletionOrEvictionStarted = o.onPodDeletionOrEvictionStarted
	}
	return o
}
```